lib/Semantic_check.ml

```diff
@@ -1,2163 +0,0 @@
-        then
-          semantic_error ~loc
-            "Target can only be accessed in the model block or in definitions \
-             of functions with the suffix _lp."
-      in
-      let returnblock = lub_originblock (List.map fst argumenttypes) in
-      match get_stan_math_function_return_type_opt uid.name argumenttypes with
-      | Some Void ->
-          semantic_error ~loc
-            ( "A returning function was expected but a non-returning function "
-            ^ uid.name ^ " was supplied." )
-      | Some (ReturnType ut) ->
-          TypedExpr
-            ( CondFunApp (uid, ues)
-            , { expr_typed_meta_origin_type= (returnblock, ut)
-              ; expr_typed_meta_loc= loc } )
-      | _ -> (
-          let _ =
-            if is_stan_math_function_name uid.name then
-              semantic_error ~loc
-                ( "Ill-typed arguments supplied to function " ^ uid.name
-                ^ ". Available signatures: "
-                ^ pretty_print_all_stan_math_function_signatures uid.name
-                ^ "\nInstead supplied arguments of incompatible type: "
-                ^ pretty_print_unsizedtypes (List.map type_of_typed_expr ues)
-                ^ "." )
-          in
-          match Symbol_table.look vm uid.name with
-          | Some (_, Fun (_, Void)) ->
-              semantic_error ~loc
-                ( "A returning function was expected but a non-returning \
-                   function " ^ uid.name ^ " was supplied." )
-          | Some (_, Fun (listedtypes, ReturnType ut)) ->
-              let _ =
-                if
-                  not
-                    (check_compatible_arguments_mod_conv uid.name listedtypes
-                       argumenttypes)
-                then
-                  semantic_error ~loc
-                    ( "Ill-typed arguments supplied to function " ^ uid.name
-                    ^ ". Available signatures:\n"
-                    ^ pretty_print_unsizedtype
-                        (Fun (listedtypes, ReturnType ut))
-                    ^ "\nInstead supplied arguments of incompatible type: "
-                    ^ pretty_print_unsizedtypes
-                        (List.map type_of_typed_expr ues)
-                    ^ "." )
-              in
-              TypedExpr
-                ( CondFunApp (uid, ues)
-                , { expr_typed_meta_origin_type= (returnblock, ut)
-                  ; expr_typed_meta_loc= loc } )
-          | Some _ ->
-              semantic_error ~loc
-                ( "A returning function was expected but a non-function value "
-                ^ uid.name ^ " was supplied." )
-          | None ->
-              semantic_error ~loc
-                ( "A returning function was expected but an undeclared \
-                   identifier " ^ uid.name ^ " was supplied." ) )
-      (* TODO: Insert informative error message in case identifier is found but not with appropriate type. *)
-      )
-  | GetLP ->
-      let _ =
-        if
```

Comment:
this indentation is crazy

Reply:
It's what ocamlformat does. How would you expect to see it indented? Are there other formatting tools that you know of?

Reply:
ocamlformat does have settings as well, we might try some of them. Is there no way to influence what it spits out? I've seen it sometimes respect my choices in certain places.